LottieSample/src/androidTest/java/com/airbnb/lottie/samples/FragmentVisibilityTests.kt

```diff
@@ -323,12 +321,17 @@ class FragmentVisibilityTests {
                         }
 
                         private fun bindLottieHolder(holder: RecyclerView.ViewHolder) {
-                            animationView = holder.itemView as LottieAnimationView
-                            (holder.itemView as LottieAnimationView).apply {
-                                repeatCount = LottieDrawable.INFINITE
-                                setAnimation(R.raw.heart)
-                                playAnimation()
-                                IdlingRegistry.getInstance().register(LottieIdlingResource(this, name = "Lottie ${Random.nextFloat()}"))
+                            if (!animationWasPlayed) {
```

Comment:
Why do you need this stateful ViewHolder?
If you want the animation to restart every time it is bound, you'd want to call play each time. If not, you can just call play in onCreate or set autoPlay to true in the xml

Reply:
Function `testPausesWhenScrolledOffScreenAndResumesWhenComesBack()` as I described in the original [issue comment](https://github.com/airbnb/lottie-android/issues/1324#issuecomment-514776627), from the declaration  - we are expecting that animation will continue playing, when we went back, by **scrolling up**, after stopping it in the middle by **scrolling down**. 

But in the scenario with **stateless** `ViewHolder` every time we **scroll up** `onBindViewHolder()`(of course if we've scrolled enough and our `ViewHolder` isn't valid anymore, in case of **10_000** scroll thats always true) callback is called and the `playAnimation()` is called also. So in any case, we aren't truly testing the  `LottieAnimation.onAttachedToWindow()` callback within its check:
```
if (autoPlay || wasAnimatingWhenDetached) {
      playAnimation();
//etc
}
```
We are just triggering another animation and testing:
` scenario.onFragment { assertTrue(it.animationView!!.isAnimating) }` 
Which doesn't make sense, as it will return `true` 100%. 

In case with **stateful** `ViewHolder` we aren't triggering animation more than one time, because of that we are verifying valid work of `LottieAnimation.onAttachedToWindow()` callback.